### PR TITLE
[chore] [pkg/pdatatest] Make exported functions consistent

### PR DESCRIPTION
**Pull request by dmitryax**
This change updates the exported functions to be consistently applied on pdata non-slice objects and called `Compare<pdata_type>`. If we want to export Compare functions for slices, we can do it later for all the slice types. The change also brings some consistency to how the errors and improves code test coverage.

The name `Compare<pdata_type>` with <pdata_type> in singular form is needed to avoid `CompareMetrics` conflict

### CODE REVIEW

File: pkg/pdatatest/plogtest/logs.go [hunk 1]
```diff
@@ -174,16 +227,21 @@ func CompareLogRecordSlices(expected, actual plog.LogRecordSlice) error {
 	}
 
 	for alr, elr := range matchingLogRecords {
-		if err := CompareLogRecords(alr, elr); err != nil {
-			return multierr.Combine(fmt.Errorf("log record with attributes: %v, does not match expected", alr.Attributes().AsRaw()), err)
+		if err := CompareLogRecord(alr, elr); err != nil {
+			return multierr.Combine(fmt.Errorf("log record with attributes %v does not match expected", alr.Attributes().AsRaw()), err)
```
Comment (djaglowski) [inline pkg/pdatatest/plogtest/logs.go:231]:
```suggestion
			return multierr.Combine(fmt.Errorf("LogRecord with attributes %v does not match expected", alr.Attributes().AsRaw()), err)
```
This seems to match the casing used elsewhere.

File: pkg/pdatatest/plogtest/logs.go [hunk 2]
```diff
@@ -96,50 +97,102 @@ func CompareLogs(expected, actual plog.Logs, options ...CompareLogsOption) error
...
+			expected.Scope().Name(), actual.Scope().Name())
+	}
+	if expected.Scope().Version() != actual.Scope().Version() {
+		return fmt.Errorf("scope version does not match expected: %s, actual: %s",
+			expected.Scope().Version(), actual.Scope().Version())
 	}
 
-	numLogRecords := expected.Len()
+	if expected.LogRecords().Len() != actual.LogRecords().Len() {
+		return fmt.Errorf("number of log records does not match expected: %d, actual: %d",
```
Comment (djaglowski) [inline pkg/pdatatest/plogtest/logs.go:180]:
```suggestion
		return fmt.Errorf("number of LogRecords does not match expected: %d, actual: %d",
```

**Review by djaglowski — COMMENTED** [review]
It's a bit difficult to follow all of the changes here. Would it make sense to split this up? Perhaps a PR for the structural changes, and another for the more naming / error message changes?

File: pkg/pdatatest/plogtest/logs.go (hunk 1 above)
Comment (dmitryax) [reply pkg/pdatatest/plogtest/logs.go:231]: `log record` is actually being used across all the logs.go. I can update it in a separate PR that would polish all the errors

**Comment by dmitryax** [comment]
> It's a bit difficult to follow all of the changes here. Would it make sense to split this up? Perhaps a PR for the structural changes, and another for the more naming / error message changes?

@djaglowski makes sense. I submitted a PR that updates only API related parts https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/17968. PTAL. I'll submit another PR with making log messages consistent

**Comment by djaglowski** [comment]
@dmitryax, thanks. That one was much easier to follow.

**Review by djaglowski — APPROVED** [review]

**Pull request merged**